R/config.R

```diff
@@ -1,45 +1,93 @@
+##' Retrieve orderly config object.
+##'
+##' @inheritParams orderly_list
+##'
+##' @return An R6 object representing the orderly config.
+##' @export
+##'
+##' @examples
+##' # The orderly demo, with lots of potential reports:
+##' path <- orderly::orderly_example("demo")
+##'
+##' orderly::orderly_config(path)
+orderly_config <- function(root = NULL, locate = TRUE) {
+  if (inherits(root, "orderly_config")) {
+    root
+  } else if (is.null(root) && locate) {
+    orderly_locate_config()
+  } else if (is.character(root)) {
+    orderly_config_$new(root)
+  } else {
+    stop("Invalid input")
+  }
+}
+
 orderly_locate_config <- function() {
   root <- find_file_descend("orderly_config.yml")
   if (is.null(root)) {
     stop("Reached root without finding 'orderly_config.yml'")
   }
-  orderly_config$new(root)
+  orderly_config_$new(root)
 }
 
 
-orderly_config_get <- function(x, locate = FALSE) {
-  if (inherits(x, "orderly_config")) {
-    x
-  } else if (is.null(x) && locate) {
-    orderly_locate_config()
-  } else if (is.character(x)) {
-    orderly_config$new(x)
-  } else {
-    stop("Invalid input")
-  }
-}
-
-
-orderly_config <- R6::R6Class(
+##' Orderly config
+##'
+##' An object representing orderly config. Reads the yml, validates
+##' it and makes the options available.
+##'
+##' @return An R6 object representing the orderly config.
+##' @export
```

Comment:
If you remove `@export` and add `@rdname orderly_config` we might get lucky and this will appear in the `orderly_config()` help without directly exposing the class